util/grpc/src/lib.rs

```diff
@@ -101,10 +99,33 @@ pub fn send_result<T>(
         ),
     }
 
-    SVC_COUNTERS.resp(&ctx, success);
+    SVC_COUNTERS.resp(&ctx, is_success);
     SVC_COUNTERS.status_code(&ctx, code);
 }
 
+/// Helper struct that provides information related to a gRPC response.
+pub struct ResponseStatus {
+    /// True if the gRPC response is ok.
+    pub is_success: bool,
+
+    /// RpcStatusCode that corresponds to the response.
+    pub code: RpcStatusCode,
+}
+
+impl ResponseStatus {
+    /// Creates a `ResponseStatus` struct from a gRPC response.
+    #[inline]
+    pub fn new<T>(response: &Result<T, RpcStatus>) -> Self {
```

Comment:
Yeah I considered `From` but I couldn't do it easily because of the generic `T` param. Maybe I'm missing an easy way to include this generic param within a `From`?

Reply:
this is what I tried locally
```rust
impl<T> From<&Result<T, RpcStatus>> for ResponseStatus {
    fn from(response: &Result<T, RpcStatus>) -> Self {
        let is_success = response.is_ok();
        let code = match response {
            Ok(_) => RpcStatusCode::OK,
            Err(e) => e.code(),
        };

        Self { is_success, code }
    }
}
```

Reply:
Done.